include/zephyr/drivers/gpio.h

```diff
@@ -216,6 +216,11 @@ extern "C" {
 					GPIO_INT_LEVELS_LOGICAL | \
 					GPIO_INT_HIGH_1)
 
+
+/** Configures GPIO interrupt to wakeup the system from low power mode.
+ */
+#define GPIO_INT_WAKEUP                (1u << 28)
```

Comment:
This PR goes on the same direction that one I already propose, see first commit of https://github.com/zephyrproject-rtos/zephyr/pull/64820.

The big differences are:
1- This define the GPIO_INT_WAKEUP symbol as an interrupt source. This means that the API that will be used is different.
2- This define the symbol as _not hidden_. I think this could open the door to add an entry at devicetree and then it is important remind the `wakeup-source` property, maybe redundant then?

IMHO this goes in the right direction and what is important is clarify the 2 points mentioned.

Reply:
Thank you. `wakeup-source` is not used in this case as we need to determine the wakeup on a per GPIO line basis which is what this PR does.

Reply:
Hi @mmahadevan108 ,

I imagine that in the future there will be intentions to bind devicetree and the wake-up pins. In that case, how do intent to do it?

Reply:
@nandojve, I have not thought of the device-tree approach.

Reply:
> @nandojve, I have not thought of the device-tree approach.

In PR https://github.com/zephyrproject-rtos/zephyr/pull/67441 I introduced binding for STM32 wake-up pins, it may be useful for other SoCs.

Reply:
@nandojve, I would like to move further on this PR. Do you have any suggestions?

Reply:
I'd like to see this flag moved to `include/zephyr/dt-bindings/gpio/gpio.h` to allow setting it via devicetree. This will allow the user/application to configure which GPIO (INT) signals should wake up the system.

Take for instance an external RTC peripheral device with an alarm INT output connected to a MCU GPIO input. On some systems, it might be desired to have this alarm INT wake up the MCU, on others it may not be desired - meaning the flag cannot be hardcoded in the RTC driver's GPIO configuration. Allowing the user/application to specify this via DTS will allow for either scenario.

Reply:
Thank you for the suggestion. I have moved the flag as suggested.